edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/modules/MetricsModule.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.Service
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using Autofac;
+    using Microsoft.Azure.Devices.Edge.Agent.Core.Metrics;
+    using Microsoft.Azure.Devices.Edge.Util.Metrics;
+    using Microsoft.Azure.Devices.Edge.Util.Metrics.NullMetrics;
+    using Microsoft.Azure.Devices.Edge.Util.Metrics.Prometheus.Net;
+
+    public class MetricsModule : Autofac.Module
+    {
+        MetricsConfig metricsConfig;
+        string iothubHostname;
+        string deviceId;
+
+        public MetricsModule(MetricsConfig metricsConfig, string iothubHostname, string deviceId)
+        {
+            this.metricsConfig = metricsConfig;
+            this.iothubHostname = iothubHostname;
+            this.deviceId = deviceId;
+        }
+
+        protected override void Load(ContainerBuilder builder)
+        {
+            builder.Register(c => new MetricsProvider("edgeagent", this.iothubHostname, this.deviceId))
+                .As<IMetricsProvider>()
+                .SingleInstance();
+
+            builder.Register(c => new Util.Metrics.Prometheus.Net.MetricsListener(this.metricsConfig.ListenerConfig, c.Resolve<IMetricsProvider>()))
+                .As<IMetricsListener>()
+                .SingleInstance();
+
+            Dictionary<Type, string> recognizedExceptions = new Dictionary<Type, string>
```

Comment:
what "string" is used for?

Reply:
That would have worked as well. However, since it's already implemented with a custom name, I don't think it needs to be changed.

Reply:
My point is this recognizedExceptions is not needed if always report exception type.  Also for any exception not defined in this recgnizedExceptions, it will report as "other".  I think exception type should be reported instead.